tests/test_lab_dev/test_states2.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for the state subpackage."""
+
+# pylint: disable=protected-access, unspecified-encoding, missing-function-docstring, expression-not-assigned, pointless-statement
+
+
+import numpy as np
+import pytest
+
+from mrmustard import settings
+
+from mrmustard.lab_dev.states import Coherent, QuadratureEigenstate
+
+# original settings
+autocutoff_max0 = settings.AUTOCUTOFF_MAX_CUTOFF
+
+
+class TestQuadratureEigenstate:
```

Comment:
Ah looks like it was the size of the file that pylint didnt like... it seems to me we want to break down states.py (and possibly transformations.py) into separate files then

Reply:
In which case for this PR maybe it would be best to have a quadrature_eigenstate.py file and test_quadrature_eigenstate.py and separate everything else out in a followup PR. Thoughts?

Reply:
Or we could do `test_ket` and `test_dm` in a separate file since they are a bit special, and keep the other "actual states" in the test_states.py

Reply:
Yeah that works remembering that the best practice is usually to name your test files accordingly to the files they're testing e.g. circuits.py and test_circuits.py to make it clear what files are being tested. So if we want to split things off between test_ket and test_dm then the states.py file should also reflect this. 

I think my issue with that is atm we only have one DM state and six (seven if including this PR) Ket states so really it would only be separating out a single state from that file. That and the fact that it's somewhat difficult to see what we do and dont have based on the .py files makes me lean toward just having dedicated files per class where we can have a dm folder and a ket folder.

Reply:
But this makes sense since ket and dm are defined in the states.base.py and all the other states are defined in states.states.py.

Reply:
Ah yes sorry you're talking about the ones that test the base classes. Yeah both of those should probably be in a separate `/test_lab_dev/test_states/base.py` (reflecting the file structure as mentioned before)

Reply:
I agree